### ISSUE TITLE
Support for application specific management groups

### ISSUE THREAD

**Issue by hglassdyb** [JPHutchins/smp#13]
Hi,

I was trying to use this library (along with `smpclient`) to test some application specific management groups. However it gave some errors when defining custom requests.

```python
class DeviceRequest(smp.message.ReadRequest):
    _GROUP_ID=65
    _COMMAND_ID=0

req = DeviceRequest()
```

```bash
$ python test_smpclient.py
Traceback (most recent call last):
  File "/test_smpclient.py", line 27, in <module>
    req = DeviceRequest()
          ^^^^^^^^^^^^^^^
  File "/.venv/lib/python3.11/site-packages/pydantic/main.py", line 176, in __init__
    self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 1 validation error for DeviceRequest
  Value error, 65 is not a valid GroupId [type=value_error, input_value={}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.7/v/value_error
```

Is there a way around this? Or am I doing something wrong?

**Comment by JPHutchins** [comment]
@hglassdyb Thanks for this!  Yes, I am working on a fix now.  If you'd like to unblock your work, you can modify smp/header.py in your venv to contain the definition for your custom group:

```python
class GroupId(IntEnum):
    OS_MANAGEMENT = 0
    IMAGE_MANAGEMENT = 1
    STATISTICS_MANAGEMENT = 2
    SETTINGS_MANAGEMENT = 3
    LOG_MANAGEMENT = 4
    RUNTIME_TESTS = 5
    SPLIT_IMAGE_MANAGEMENT = 6
    TEST_CRASH = 7
    FILE_MANAGEMENT = 8
    SHELL_MANAGEMENT = 9
    ZEPHYR = 63
    _APPLICATIION_CUSTOM_MIN = 64
    INTERCREATE = 64
    MY_GROUP = 65
```

The fix will allow inheritance the way you've defined it.

**Comment by JPHutchins** [comment]
@hglassdyb This should be fixed in 0.3.4, test it out at your convenience, thanks!

**Comment by JPHutchins** [comment]
Good point!  I need to move 64 out of the GroupId enum, it's attempting to validate the CommandIds.

```python
@unique
class GroupId(IntEnum):
    OS_MANAGEMENT = 0
    IMAGE_MANAGEMENT = 1
    STATISTICS_MANAGEMENT = 2
    SETTINGS_MANAGEMENT = 3
    LOG_MANAGEMENT = 4
    RUNTIME_TESTS = 5
    SPLIT_IMAGE_MANAGEMENT = 6
    TEST_CRASH = 7
    FILE_MANAGEMENT = 8
    SHELL_MANAGEMENT = 9
    ZEPHYR = 63


class CustomGroupId(IntEnum):
    INTERCREATE = 64
```
